### PR TITLE
Remove uneeded clones now that TrustedStep implies Copy

### PR DESCRIPTION
**Pull request by krtab**
This is a follow up to 11fa1764ee4819aa674ca861c5e9a8fafd7a59e6 (from #112083)

### CODE REVIEW

File: library/core/src/iter/range.rs [hunk 1]
```diff
@@ -1224,15 +1224,15 @@ impl<T: TrustedStep> RangeInclusiveIteratorImpl for ops::RangeInclusive<T> {
 
         while self.start < self.end {
             // SAFETY: just checked precondition
-            let n = unsafe { Step::backward_unchecked(self.end.clone(), 1) };
+            let n = unsafe { Step::backward_unchecked(self.end().clone(), 1) };
```
Comment (cuviper) [inline library/core/src/iter/range.rs:1227]: What happened here? Shouldn't this be:
```suggestion
            let n = unsafe { Step::backward_unchecked(self.end, 1) };
```

File: library/core/src/iter/range.rs [hunk 2]
```diff
@@ -1224,15 +1224,15 @@ impl<T: TrustedStep> RangeInclusiveIteratorImpl for ops::RangeInclusive<T> {
...
             let n = mem::replace(&mut self.end, n);
             accum = f(accum, n)?;
         }
 
         self.exhausted = true;
 
         if self.start == self.end {
-            accum = f(accum, self.start.clone())?;
+            accum = f(accum, self.start().clone())?;
```
Comment (cuviper) [inline library/core/src/iter/range.rs:1235]: And similarly:
```suggestion
            accum = f(accum, self.start)?;
```

**Comment by krtab** [comment]
Wow, thanks for catching that @cuviper, that's what happen when you open PR on a Friday afternoon I guess.

**Comment by cuviper** [comment]
r? cuviper
@bors r+ rollup

**Pull request merged**